jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/internal/cdi/ManagedBeanDiagnosticsParticipant.java

```diff
@@ -382,6 +387,14 @@ private void invalidParamsCheck(JavaDiagnosticsContext context, String uri, ICom
                 for (String annotation : paramScopes) {
                     invalidAnnotations.add("@" + DiagnosticUtils.getSimpleName(annotation));
                 }
+                
+                if (paramScopes != null && !paramScopes.isEmpty()) {
```

Comment:
1. `paramScopes` is never null. 2. I doubt checking for empty saves much time. 3. In the state of interest, `paramScopes` should be a list of the three mutually exclusive annotations possibly in fully qualified names form. If not, we know the names have been validated by `DiagnosticUtils.getMatchedJavaElementNames()`. In this case, normalizing the names to short names then eliminating duplicates in the Set should be good enough to determine if the full set of names in INVALID_INITIALIZER_PARAMS_FQ has been used on the parameter. Comparing with `invalidAnnotations.equals(expected)` should not be needed. You would need to test with a combination of short and FQ names and repeated and unique annotations to be sure.